src/js/base/runtime.js

```diff
@@ -475,6 +473,18 @@ function (Namespace, jsnums, codePoint, util, exnStackParser, loader, seedrandom
       return thisRuntime.getFieldLoc(obj, field, ["runtime"]);
     }
 
+    function getBracket(loc, obj, field) {
+      checkArityC(loc, 3, arguments, false);
+      if (obj && obj.dict && obj.dict["get-value"]) {
+        return getFieldLoc(obj, "get-value", loc).app(field);
```

Comment:
We could call isFunction on the result of the getField here for another `BadBracketException` case. As-is this will apply undefined for `{get-value: "not-a-function"}`